<a href="https://colab.research.google.com/github/andrewsundberg/DS301/blob/master/Lab9_DNN_Image_Classifier_SundbergAndrew.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
try:
  # %tensorflow_version only exists in Colab.
  %tensorflow_version 2.x
except Exception:
  pass

TensorFlow 2.x selected.


In [2]:
from __future__ import absolute_import, division, print_function, unicode_literals

# TensorFlow and tf.keras
import tensorflow as tf
from tensorflow import keras

# Helper libraries
import numpy as np
import matplotlib.pyplot as plt

print(tf.__version__)

2.0.0


In [3]:
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/drive


In [0]:
#drive.mount("/content/drive", force_remount=True)

In [5]:
from keras import backend as K

Using TensorFlow backend.


In [0]:
img_width, img_height = 300,300


In [0]:
nb_train_samples = 1027
nb_validation_samples = 256
epochs = 50
batch_size = 16

In [0]:

if K.image_data_format() == 'channels_first':
    input_shape = (3, img_width, img_height)
else:
    input_shape = (img_width, img_height, 3)

In [0]:
from keras.models import Sequential
#from keras.layers import Conv2D, MaxPooling2D
from keras.layers import Activation, Dropout, Flatten, Dense

model = keras.Sequential([
    keras.layers.Flatten(input_shape=(256, 256,3)),
    keras.layers.Dense(128, activation='relu'),
    keras.layers.Dense(10, activation='softmax')
])

In [0]:
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

In [0]:
from keras_preprocessing.image import ImageDataGenerator
datagen=ImageDataGenerator()
train_datagen = ImageDataGenerator(
        rescale=1./255)
test_datagen = ImageDataGenerator(rescale=1./255)



In [11]:
ls

drive/  sample_data/


In [12]:
cd drive/


/content/drive


In [13]:
ls

'My Drive'/


In [14]:
cd My\ Drive

/content/drive/My Drive


In [15]:
ls

'2018 Western Conference Finals, Warriors @ Rockets.gsheet'
 AaronRodgersAaronJonesDoubleTrouble-min.png
 AaronRodgersAaronJonesDoubleTrouble.png
'Andrew Sundberg Resume.gdoc'
'Career Fair Reflection.gdoc'
 CBB2018.gsheet
'Chapter 14: Listening in Partnerships.gdoc'
'Colab Notebooks'/
'COMST 102 Paper 1.gdoc'
 cr_class.py
 cr_class.py.gdoc
 dataset/
 Dogs.gdoc
'DS201X Exam 1 Sample questions.gdoc'
'English 150'/
'English 250'/
'Fall 2016 Lifeguard Survey.docx'
'Fall 2016 Lifeguard Survey.docx.gdoc'
'Geology 100.gdoc'
'Homework 1 (1).gsheet'
'Homework 1.gsheet'
'Homework 1.xlsx'
'Homework Seven Write-up.docx'
'Homework Seven Write-up.docx.gdoc'
'Internship curriculum 11-30-16.docx'
'John Deere Cover Letter.gdoc'
'KIC Document 0001.pdf'
'Lab 2.xlsx'
'Mixtape (2).png'
 Mixtape.png
 Mixtape.xcf
 Most-Recent-Cohorts-All-Data-Elements.xlsx
'My Route.docx'
'My Route.docx.gdoc'
'Nicholas Brothers.gslides'
'Offline Homework 2.docx'
'Offline Homework 2.gdoc'
'personnafax--career analysis (9).pdf

In [16]:
cd dataset/

/content/drive/My Drive/dataset


In [17]:
ls

 dogs-cats-images/  'dogs-cats-images (1)'/   horses-or-humans-dataset/


In [18]:
cd horses-or-humans-dataset/

/content/drive/My Drive/dataset/horses-or-humans-dataset


In [19]:
ls

train/  validation/


In [20]:
pwd

'/content/drive/My Drive/dataset/horses-or-humans-dataset'

In [21]:
#Sample images: Horses or Humans gerated images
#train_data = train_datagen.flow_from_directory(r'/content/drive/My Drive/dataset/horses-or-humans-dataset/train',class_mode='binary')
#test_data=test_datagen.flow_from_directory(r'/content/drive/My Drive/dataset/horses-or-humans-dataset/validation',class_mode='binary')


Found 1027 images belonging to 2 classes.
Found 256 images belonging to 2 classes.


In [23]:
#Your task, use Dogs or Cats images instead.
train_data = train_datagen.flow_from_directory(r'/content/drive/My Drive/dataset/dogs-cats-images/training_set',class_mode='binary')
test_data=test_datagen.flow_from_directory(r'/content/drive/My Drive/dataset/dogs-cats-images/test_set',class_mode='binary')

Found 8000 images belonging to 2 classes.
Found 2000 images belonging to 2 classes.


In [24]:
model.fit_generator(
    train_data,
    steps_per_epoch=nb_train_samples // batch_size/8,
    epochs=20,
    validation_data=test_data,
    validation_steps=nb_validation_samples // batch_size)

Epoch 1/20
8/8 [==============================] - 354s 44s/step - loss: 6.8660 - accuracy: 0.4766 - val_loss: 7.7128 - val_accuracy: 0.5215
Epoch 2/20
8/8 [==============================] - 4s 511ms/step - loss: 8.3739 - accuracy: 0.4805 - val_loss: 7.7128 - val_accuracy: 0.5215
Epoch 3/20
8/8 [==============================] - 69s 9s/step - loss: 8.0590 - accuracy: 0.5000 - val_loss: 7.7128 - val_accuracy: 0.5215
Epoch 4/20
8/8 [==============================] - 115s 14s/step - loss: 8.2479 - accuracy: 0.4883 - val_loss: 7.7128 - val_accuracy: 0.5215
Epoch 5/20
8/8 [==============================] - 113s 14s/step - loss: 7.8702 - accuracy: 0.5117 - val_loss: 7.7128 - val_accuracy: 0.5215
Epoch 6/20
8/8 [==============================] - 106s 13s/step - loss: 8.9405 - accuracy: 0.4453 - val_loss: 7.7128 - val_accuracy: 0.5215
Epoch 7/20
8/8 [==============================] - 116s 15s/step - loss: 7.5554 - accuracy: 0.5312 - val_loss: 7.7128 - val_accuracy: 0.5215
Epoch 8/20
8/8 [======

I changed the batch size and epachs and the accuracy is only 0.5215.